libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/AnnotationsRefresher.kt

```diff
@@ -0,0 +1,81 @@
+package com.mapbox.navigation.base.internal.route
+
+import com.mapbox.api.directions.v5.models.LegAnnotation
+import com.mapbox.navigation.utils.internal.logW
+import kotlin.math.min
+
+internal object AnnotationsRefresher {
+
+    private const val TAG = "AnnotationsRefresher"
+
+    fun getRefreshedAnnotations(
+        oldAnnotation: LegAnnotation?,
+        newAnnotation: LegAnnotation?,
+        legGeometryIndex: Int
+    ): LegAnnotation? {
+        if (newAnnotation == null) {
+            return null
+        }
+        val congestionNumeric = mergeAnnotationProperty(
+            oldAnnotation,
+            newAnnotation,
+            legGeometryIndex,
+        ) { congestionNumeric() }
+        val congestion = mergeAnnotationProperty(
+            oldAnnotation,
+            newAnnotation,
+            legGeometryIndex,
+        ) { congestion() }
+        val distance = mergeAnnotationProperty(
+            oldAnnotation,
+            newAnnotation,
+            legGeometryIndex,
+        ) { distance() }
+        val duration = mergeAnnotationProperty(
+            oldAnnotation,
+            newAnnotation,
+            legGeometryIndex,
+        ) { duration() }
+        val speed = mergeAnnotationProperty(
+            oldAnnotation,
+            newAnnotation,
+            legGeometryIndex,
+        ) { speed() }
+        val maxSpeed = mergeAnnotationProperty(
+            oldAnnotation,
+            newAnnotation,
+            legGeometryIndex,
+        ) { maxspeed() }
+        // unrecognized properties migrate from new annotation
```

Comment:
Good thinking :+1: it's a gnarly piece of code though, do we have a test for it?

Reply:
Of course, the last one in the `AnnotationsRefresherTest`.

Reply:
Perfect :+1: